app/main.py

```diff
@@ -1,34 +1,69 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self,
+                 row: int,
+                 column: int,
+                 is_alive: bool = True
+                 ) -> None:
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self,
+                 start: tuple,
+                 end: tuple,
+                 is_drowned: bool = False
+                 ) -> None:
+        self.start = start
+        self.end = end
+        self.decks = self.make_decks_for_ship()
+        self.is_drowned = is_drowned
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def make_decks_for_ship(self) -> list:
+        decks = []
+        ship_length = self.end[0] - self.start[0] + self.end[1] - self.start[1]
+        for i in range(0, ship_length + 1):
+            if self.start[0] == self.end[0]:
+                decks.append(Deck(self.start[0], self.start[1] + i))
+            else:
+                decks.append(Deck(self.start[0] + i, self.start[1]))
```

Comment:
Instead of some calculation and if statements just use for loop

Reply:
Thank you